gobblin-service/src/main/java/org/apache/gobblin/service/modules/dataset/HiveDatasetDescriptor.java

```diff
@@ -65,16 +67,28 @@ public HiveDatasetDescriptor(Config config) throws IOException {
       partitionFormat = "";
       conflictPolicy = HiveCopyEntityHelper.ExistingEntityPolicy.REPLACE_TABLE.name();
     }
+
+    whitelistBlacklist = new WhitelistBlacklist(config.withValue(WhitelistBlacklist.WHITELIST,
+        ConfigValueFactory.fromAnyRef(createHiveDatasetWhitelist())));
     this.setRawConfig(this.getRawConfig()
         .withValue(CONFLICT_POLICY, ConfigValueFactory.fromAnyRef(conflictPolicy))
         .withValue(PARTITION_COLUMN, ConfigValueFactory.fromAnyRef(partitionColumn))
         .withValue(PARTITION_FORMAT, ConfigValueFactory.fromAnyRef(partitionFormat))
-        .withValue(WHITELIST_TABLES, ConfigValueFactory.fromAnyRef(createWhitelistedTables())
+        .withValue(HiveDatasetFinder.HIVE_DATASET_PREFIX + "." + WhitelistBlacklist.WHITELIST,
+            ConfigValueFactory.fromAnyRef(createHiveDatasetWhitelist())
         ));
   }
 
-  private String createWhitelistedTables() {
-    return this.tableName.replace(',', '|');
+  // If the db name contains wildcards, whitelist is created as <regex_db>.*
+  // Otherwise, whitelist is created as <db>.tables.
+  // This is the format which HiveDatasetFinder understands.
+  // e.g. db=ei_tracking, table=zephyr*,abook*, whitelist will be ei_tracking.zephyr*|abook*
+  String createHiveDatasetWhitelist() {
+    if (new GlobPattern(this.databaseName).hasWildcard()) {
```

Comment:
Maybe add a comment why we are using Hadoop's GlobPattern instead of java.util.regex?

Reply:
I am using Hadoop's GlobPattern instead of java.util.regex, because I could not find any API java.util.regex which tells if the string is a plain string or contains special characters. Do you know any API in java.util.regex? or should I put this reason in comment?

Reply:
Yeah - would be good to add that as a comment.